tt_metal/hostdevcommon/common_runtime_address_map.h

```diff
@@ -43,6 +43,36 @@ constexpr static std::uint32_t PRINT_BUFFER_T1 = PRINT_BUFFER_T0 + PRINT_BUFFER_
 constexpr static std::uint32_t PRINT_BUFFER_T2 = PRINT_BUFFER_T1 + PRINT_BUFFER_SIZE; // TRISC2
 constexpr static std::uint32_t PRINT_BUFFER_BR = PRINT_BUFFER_T2 + PRINT_BUFFER_SIZE; // BRISC
 
+constexpr static std::uint32_t PROFILER_L1_MARKER_UINT32_SIZE = 2;
+constexpr static std::uint32_t PROFILER_L1_MARKER_BYTES_SIZE = PROFILER_L1_MARKER_UINT32_SIZE * sizeof(uint32_t);
+
+constexpr static std::uint32_t PROFILER_L1_PROGRAM_ID_COUNT = 2;
+constexpr static std::uint32_t PROFILER_L1_GUARANTEED_MARKER_COUNT = 4;
+constexpr static std::uint32_t PROFILER_L1_OPTIONAL_MARKER_COUNT = 250;
+
+constexpr static std::uint32_t PROFILER_L1_VECTOR_SIZE = (PROFILER_L1_OPTIONAL_MARKER_COUNT + PROFILER_L1_GUARANTEED_MARKER_COUNT + PROFILER_L1_PROGRAM_ID_COUNT) * PROFILER_L1_MARKER_UINT32_SIZE;
+constexpr static std::uint32_t PROFILER_L1_BUFFER_SIZE = PROFILER_L1_VECTOR_SIZE  * sizeof(uint32_t);
+constexpr static std::uint32_t PROFILER_L1_CONTROL_VECTOR_SIZE = 32;
+constexpr static std::uint32_t PROFILER_L1_CONTROL_BUFFER_SIZE = PROFILER_L1_CONTROL_VECTOR_SIZE * sizeof(uint32_t);
+
+constexpr static std::uint32_t PROFILER_L1_BUFFER_BR = 108 * 1024;
+constexpr static std::uint32_t PROFILER_L1_BUFFER_NC = PROFILER_L1_BUFFER_BR + PROFILER_L1_BUFFER_SIZE;
+constexpr static std::uint32_t PROFILER_L1_BUFFER_T0 = PROFILER_L1_BUFFER_NC + PROFILER_L1_BUFFER_SIZE;
+constexpr static std::uint32_t PROFILER_L1_BUFFER_T1 = PROFILER_L1_BUFFER_T0 + PROFILER_L1_BUFFER_SIZE;
+constexpr static std::uint32_t PROFILER_L1_BUFFER_T2 = PROFILER_L1_BUFFER_T1 + PROFILER_L1_BUFFER_SIZE;
+constexpr static std::uint32_t PROFILER_L1_BUFFER_CONTROL = PROFILER_L1_BUFFER_T2 + PROFILER_L1_BUFFER_SIZE;
+
+constexpr static std::uint32_t PROFILER_L1_END_ADDRESS = PROFILER_L1_BUFFER_CONTROL + PROFILER_L1_CONTROL_BUFFER_SIZE;
+
+constexpr static std::uint32_t PROFILER_OP_SUPPORT_COUNT = 1100;
+constexpr static std::uint32_t PROFILER_FULL_HOST_VECTOR_SIZE_PER_RISC = PROFILER_L1_MARKER_UINT32_SIZE * (PROFILER_L1_PROGRAM_ID_COUNT +  PROFILER_L1_GUARANTEED_MARKER_COUNT) * PROFILER_OP_SUPPORT_COUNT;
+constexpr static std::uint32_t PROFILER_FULL_HOST_BUFFER_SIZE_PER_RISC = PROFILER_FULL_HOST_VECTOR_SIZE_PER_RISC * sizeof(uint32_t);
+
+constexpr static std::uint32_t PROFILER_RISC_COUNT = 5;
+
+static_assert (PROFILER_FULL_HOST_BUFFER_SIZE_PER_RISC > PROFILER_L1_BUFFER_SIZE);
+
```

Comment:
you don't need to change this PR, but fyi, I'm trying to deprecate this file.  going forward, I want the device to export capabilities (using dev_msg and dev_mem_map) and then have everything else be dynamic (eg, semaphore locations, etc).

Reply:
For sure, I can certainly contribute to the effort. Profiler related addresses should be straight forward to port over.